### DISCUSSION TITLE
Setting purpose and creating collections

### DISCUSSION THREAD

**Discussion by sanders3d** [Autodesk/maya-usd#1628] [General]
Hello,
I'm trying to create collections from a model that I have magically created. And I'm trying to replicate the workflow that I have in Houdini. 

A few things I have figured out

I've imported the stage (model) from a usd file, created a new sublayer (to be saved out. I've changed the prims that I wanted to be either render/proxy purpose
![image](https://user-images.githubusercontent.com/44543707/129610560-708b5ecc-1e3c-44af-9930-a0e477d0a564.png)

and I'm now ready to create some collections to be used... and this is where everything stops. I'm also not sure how to add shaders to the prims after I've exported the model to USD.

Any pointers? 

regards
stefan

**Comment by wtelford** [comment]
Hey Stefan,
We don't yet have a native USD material authoring workflow in place. Materials built on the Maya-side will export to USD though. We are currently working on workflows around in-memory data translation, which would allow you to pull USD items in memory to the maya data model, modify them (apply or modify a material for example), and then push them back to USD.

This is already possible with our released plugins through scripting, but we are looking to wrap the functionality in a nicer workflow. You can see an example of the sccripted implementation here https://github.com/Autodesk/maya-usd/discussions/1054#discussioncomment-641407

In regards to collection authoring, we are currently investigating "isolate select" behavior for USD data in Maya. On the Maya side this utilizes objectSets, so as part of that investigation we are looking at whether we will use the objectSet implementation on the USD side....and whether or not objectSets would map to collections in USD.

**Reply by sanders3d** [reply]
Thanks for the answers!

**Comment by Xaia** [comment]
Hi @wtelford 
Just wanted to chip in regarding Materials. This is the current state and I don't think any of those are working. 
![usd_materials](https://user-images.githubusercontent.com/60515654/129924094-7f105e24-42b0-4f8e-b398-2a6f3b9168b7.png)

Cheers
Xaia

**Reply by JGamache-autodesk** [reply]
Yes, indeed the rendering options are limited at the moment:

- There is no concept of a color correction node in the [UsdPreviewSurface specification](https://graphics.pixar.com/usd/docs/UsdPreviewSurface-Proposal.html), so there is no current way of exporting either CC or MayaCC nodes to that shading context. They end up getting skipped.
- We have started implementing the step above, which is a MaterialX render context, where we could export those nodes assuming we have a MaterialX node definition in the library that matches the expected color correction semantics. This project is currently in the very early stages and there is a lot of export code still missing to get the expected results. We will gladly accept any contribution that increases the export coverage of Maya shading nodes.
